meetings/2021-06-16.md

```diff
@@ -0,0 +1,150 @@
+  * the Guard on the server — authorizing access to the resource
+  * the Guard near the wallet (essentially the Launcher App) — authorizing when to sign headers, give out tokens, or provide credentials
+  
+Henry: on the one side, you may have a wallet; on the other, you verify the credential. Is that what UMA is trying to do? 
+
+Aaron: First of all, if you have an authorization server like UMA (just a specialization of OAuth 2), you can actually dispense with the authorization guard on your AS. It leads to a much cleaner separation of concerns. All you need is that the token would be valid for one specific request.
+
+Henry: I am very skeptical about this. There are reasons to be skeptical, as OAuth hardly works for our use case of decentralized hyper apps fetching data by following links. (In fact, it seems to me more like a catastrophe, a point that was echoed by some in last week's discussion on the Credentials CG). This is requiring me to have to read UMA and analyze why it might be wrong or if it is right somehow. That's a lot of work.
+
+Aaron: I think you've clearly articulated your position, thank you. OAuth is a very general authorization framework that has worked quite successfully and is widely deployed on the web today.
+
+Henry: It doesn't solve the problem we're trying to address. It solves the problem of Facebook asking people to enter their gmail password to allow FB to contact their friends and invite them to their social network.
+
+Aaron: Henry, you are conflating OIDC and OAuth2. We're talking about OAuth2, not OIDC.
+
+Henry: one does not need in depth knowledge of either technology to understand their limitations, which arise directly out of not accepting the Semantic web or linked data principles. By deliberately ignoring these technologies, they have had to come up with very awkward answers to problems that would have had clean mathematical/logical solutions otherwise.
+
+Henry: You say we don't need the Guard. Do we now throw out the authz frameworks we have been building?
+
+Aaron: Access Control is completely orthogonal to this. WAC and ACP are data models. UMA is a protocol.
+
+Henry: WAC is actually a protocol:  The web is designed so you can get to a resource from anywhere. So you reach a resource, get a 401, and follow the link header to the Access Control Resource. Then decide which credential(s) to send, if any, and the server can authorize the request.
+
+Aaron: Both WAC and ACP are orthogonal. The Guard on the server level runtime is no longer needed, because all they need to know is: "Is the token valid for this request?" The authorization server goes to `/foo/bar`, gets a 401. (??? @bblfish: @acoburn - something missing here) UMA defines you take the value to your Authz server which gives you a token for that particular request, and that's where ACP or WAC gets evaluated.
```

Comment:
```suggestion
Aaron: Both WAC and ACP are orthogonal. The Guard on the server level runtime is no longer needed, because all they need to know is: "Is the token valid for this request?" The app goes to `/foo/bar`, gets a 401 with a WWW-Authenticate header which tells a client what to do next. For example: use this OAuth2 authorization server or that UMA server or something else. The app then interacts with the authorization server whose protocol it understands and that AS produces an access token for that particular request. It is during this access token creation process that ACP or WAC is evaluated.```